clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminApiHandler.java

```diff
@@ -70,6 +71,22 @@
      */
     ApiResult<K, V> handleResponse(Node broker, Set<K> keys, AbstractResponse response);
 
+    /**
+     * Callback that is invoked when a request hits an UnsupportedVersionException. If the
+     * exception can be handled and the request retried, the appropriate changes to the request
+     * should be applied and that should be indicated via the returned boolean value. If the
+     * exception is thrown during the lookup stage the handler should delegate the handling to the
+     * lookup strategy.
+     *
+     * @return True if the exception can be handled; false otherwise.
+     */
+    default boolean handleUnsupportedVersionException(
```

Comment:
I forgot the details about the whole call chain, but I think we should decide which keys (if any) can retry with lower version and which keys can not, keys which are not retriable should be put into the result, and others are left out of the result which will be retried automatically, similar to what we do in `handleResponse`.

Reply:
Done.